spec/rspec/support/spec/stderr_splitter_spec.rb

```diff
@@ -101,26 +101,28 @@
 
   # This spec replicates what matchers do when capturing stderr, e.g `to_stderr_from_any_process`
   it 'is able to restore the stream from a cloned StdErrSplitter' do
-    if RSpec::Support::Ruby.jruby?
-      skip """
-      This spec is currently unsupported on JRuby on CI due to tempfiles not being
-      a file, this situtation was discussed here https://github.com/rspec/rspec-support/pull/598#issuecomment-2200779633
-      """
+    if RSpec::Support::Ruby.jruby? && !RSpec::Support::Ruby.jruby_9000?
+      skip "JRuby < 9.0.0.0 doesn't have a predictable identifier for stdout"
     end
 
     cloned = splitter.clone
-    expect(splitter.to_io).not_to be_a(File)
+    expect(splitter.to_io).to be_stderr
 
     tempfile = Tempfile.new("foo")
     begin
       splitter.reopen(tempfile)
-      expect(splitter.to_io).to be_a(File)
+      expect(splitter.to_io).to_not be_stderr
```

Comment:
@nevinera Unless I'm mistaken this represents the same effect as what you were testing, I'm not sure if I will merge this yet but it at least works for some JRuby

Reply:
Ah, I think the part I didn't follow was the bottom method - I assumed `be_stderr` was calling a `stderr?` method I wasn't aware of on the stream/splitter >.<

I'm not confident about how consistent that `inspect` is, but I'll trust your plan :-)